src/v2/providers/identity.ts

```diff
@@ -0,0 +1,233 @@
+    | Promise<void>
+): BlockingFunction;
+export function beforeUserCreated(
+  opts: BlockingOptions,
+  handler: (
+    event: AuthBlockingEvent
+  ) =>
+    | BeforeCreateResponse
+    | Promise<BeforeCreateResponse>
+    | void
+    | Promise<void>
+): BlockingFunction;
+export function beforeUserCreated(
+  optsOrHandler:
+    | BlockingOptions
+    | ((
+        event: AuthBlockingEvent
+      ) =>
+        | BeforeCreateResponse
+        | Promise<BeforeCreateResponse>
+        | void
+        | Promise<void>),
+  handler?: (
+    event: AuthBlockingEvent
+  ) =>
+    | BeforeCreateResponse
+    | Promise<BeforeCreateResponse>
+    | void
+    | Promise<void>
+): BlockingFunction {
+  return beforeOperation('beforeCreate', optsOrHandler, handler);
+}
+
+/**
+ * Handle an event that is triggered before a user is signed in.
+ */
+export function beforeUserSignedIn(
+  handler: (
+    event: AuthBlockingEvent
+  ) =>
+    | BeforeSignInResponse
+    | Promise<BeforeSignInResponse>
+    | void
+    | Promise<void>
+): BlockingFunction;
+export function beforeUserSignedIn(
+  opts: BlockingOptions,
+  handler: (
+    event: AuthBlockingEvent
+  ) =>
+    | BeforeSignInResponse
+    | Promise<BeforeSignInResponse>
+    | void
+    | Promise<void>
+): BlockingFunction;
+export function beforeUserSignedIn(
+  optsOrHandler:
+    | BlockingOptions
+    | ((
+        event: AuthBlockingEvent
+      ) =>
+        | BeforeSignInResponse
+        | Promise<BeforeSignInResponse>
+        | void
+        | Promise<void>),
+  handler?: (
+    event: AuthBlockingEvent
+  ) =>
+    | BeforeSignInResponse
+    | Promise<BeforeSignInResponse>
+    | void
+    | Promise<void>
+): BlockingFunction {
+  return beforeOperation('beforeSignIn', optsOrHandler, handler);
+}
+
+/** @internal */
+export function beforeOperation(
+  eventType: AuthBlockingEventType,
+  optsOrHandler:
+    | BlockingOptions
+    | ((
+        event: AuthBlockingEvent
+      ) =>
+        | BeforeCreateResponse
+        | BeforeSignInResponse
+        | void
+        | Promise<BeforeCreateResponse>
+        | Promise<BeforeSignInResponse>
+        | Promise<void>),
+  handler: (
+    event: AuthBlockingEvent
+  ) =>
+    | BeforeCreateResponse
+    | BeforeSignInResponse
+    | void
+    | Promise<BeforeCreateResponse>
+    | Promise<BeforeSignInResponse>
+    | Promise<void>
+): BlockingFunction {
+  if (!handler || typeof optsOrHandler === 'function') {
+    handler = optsOrHandler as (
+      event: AuthBlockingEvent
+    ) =>
+      | BeforeCreateResponse
+      | BeforeSignInResponse
+      | void
+      | Promise<BeforeCreateResponse>
+      | Promise<BeforeSignInResponse>
+      | Promise<void>;
+    optsOrHandler = {};
+  }
+
+  const { opts, accessToken, idToken, refreshToken } = getOpts(
+    optsOrHandler as BlockingOptions
+  );
+
+  const func: any = wrapHandler(eventType, handler, true);
+
+  const legacyEventType = `providers/cloud.auth/eventTypes/user.${eventType}`;
```

Comment:
q: Is there non-legacy event type that's coming out soon?

Reply:
It's been thrown around that if v2 Auth Blocking functions get EventArc support, then we could update this event type